options.go

```diff
@@ -115,13 +114,24 @@ func HashRingConfig(c *hashring.Configuration) Option {
 	}
 }
 
-// Logger is used to specify a bark-compatible logger that will be used for
-// all Ringpop logging. If a logger is not provided, one will be created
-// automatically.
-func Logger(l log.Logger) Option {
+// Logger is used to specify a bark-compatible logger that will be used for all
+// Ringpop logging and the log levels for all modules. If a logger is not
+// provided, one will be created automatically.
+func Logger(conf logger.Options) Option {
```

Comment:
This breaks backwards-compatibility, which I'd rather not do.

Could we solve this by having `Logger` and `LoggingConfig` as separate options? It matches the `HashRingConfig` naming pattern. And we could just deprecate `Logger` later, if we really want.